pkg/tsdb/prometheus/framing_test.go

```diff
@@ -4,71 +4,40 @@ import (
 	"bytes"
 	"context"
 	"encoding/json"
-	"fmt"
 	"io"
 	"net/http"
 	"os"
 	"path/filepath"
 	"testing"
 	"time"
 
+	"github.com/stretchr/testify/require"
+
 	"github.com/grafana/grafana-plugin-sdk-go/backend"
 	"github.com/grafana/grafana-plugin-sdk-go/experimental"
 	"github.com/grafana/grafana/pkg/infra/tracing"
 	"github.com/prometheus/client_golang/api"
 	apiv1 "github.com/prometheus/client_golang/api/prometheus/v1"
-	"github.com/stretchr/testify/require"
 )
 
 func TestMatrixResponses(t *testing.T) {
 	t.Run("parse a simple matrix response", func(t *testing.T) {
-		err := testScenario("range_simple")
-		require.NoError(t, err)
+		testScenario(t, "range_simple")
 	})
 
 	t.Run("parse a simple matrix response with value missing steps", func(t *testing.T) {
-		err := testScenario("range_missing")
-		require.NoError(t, err)
+		testScenario(t, "range_missing")
 	})
 
 	t.Run("parse a response with Infinity", func(t *testing.T) {
-		err := testScenario("range_infinity")
-		require.NoError(t, err)
+		testScenario(t, "range_infinity")
 	})
 
 	t.Run("parse a response with NaN", func(t *testing.T) {
-		err := testScenario("range_nan")
-		require.NoError(t, err)
+		testScenario(t, "range_nan")
 	})
 }
 
-// when memory-profiling this benchmark, these commands are recommended:
-// - go test -benchmem -run=^$ -benchtime 1x -memprofile memprofile.out -memprofilerate 1 -bench ^BenchmarkJson$ github.com/grafana/grafana/pkg/tsdb/prometheus
-// - go tool pprof -http=localhost:6061 memprofile.out
-func BenchmarkJson(b *testing.B) {
```

Comment:
Moved the benchmark to it's own file. I also removed the error checking in the benchmark because the test's only job is to measure performance rather than correctness.